### PR TITLE
DAOS-2383 control: drop privileges before io_server start-up

### PR DESCRIPTION
**Pull request by tanabarr**
daos_server requires start-up in root/privileged mode to perform
tasks such as format. After control plane setup and configurations
tasks and before forking io_server, drop privileges by changing
effective uid/gid to that of the user provided in config file.
io_server will then be launched by normal user.

Signed-off-by: Tom Nabarro <tom.nabarro@intel.com>

### CODE REVIEW

File: src/control/server/config_types.go [hunk 1]
```diff
@@ -227,6 +227,7 @@ type configuration struct {
 	NrHugepages    int             `yaml:"nr_hugepages"`
 	ControlLogMask ControlLogLevel `yaml:"control_log_mask"`
 	ControlLogFile string          `yaml:"control_log_file"`
+	Username       string          `yaml:"username"`
```
Comment (dpquigl) [inline src/control/server/config_types.go]: Admins might want to run the program with a separate gid as well. I would put User/Group as the config options.

File: src/control/server/main.go [hunk 2]
```diff
@@ -130,6 +174,14 @@ func serverMain() error {
 		return err
 	}
 
+	if syscall.Getuid() == 0 {
+		log.Debugf("Dropping privileges...")
+		if err := drop(config.Username); err != nil {
+			log.Errorf("Failed to drop privileges: %s", err)
+			// TODO: don't continue as root
```
Comment (dpquigl) [inline src/control/server/main.go]: I'm going to add an --insecure flag with the gRPC stuff so maybe we make it take a string of things that we'll allow to be insecure and make this an option to not drop privileges from root?

Comment (tanabarr) [reply src/control/server/main.go]: can't we make that change when you add the insecure flag? seems a bit premature to add that capability in this patch

File: src/control/server/config_types.go (hunk 1 above)
Comment (tanabarr) [reply src/control/server/config_types.go]: sure

**Pull request merged**